redisinsight/ui/src/pages/vector-search/query/VectorSearchQuery.tsx

```diff
@@ -1,11 +1,92 @@
 import React from 'react'
+import {
+  ResizablePanel,
+  ResizablePanelHandle,
+} from 'uiSrc/components/base/layout'
+import QueryWrapper from 'uiSrc/pages/workbench/components/query'
+import { HIDE_FIELDS } from 'uiSrc/components/query/query-card/QueryCardHeader/QueryCardHeader'
+import {
+  StyledNoResultsWrapper,
+  StyledResizableContainer,
+} from './VectorSearchQuery.styles'
 import { HeaderActions } from './HeaderActions'
+import { useQuery } from './useQuery'
 import { CreateIndexWrapper } from '../create-index/styles'
+import CommandsViewWrapper from '../components/commands-view'
 
-// TODO: implement this component
-// https://www.figma.com/design/oO2eYRuuLmfzUYLkvCkFhM/Search-page?node-id=645-37412&t=TSwcttCYa4Ld9WzC-4
-export const VectorSearchQuery = () => (
-  <CreateIndexWrapper direction="column" justify="between">
-    <HeaderActions />
-  </CreateIndexWrapper>
-)
+export const VectorSearchQuery = () => {
+  const {
+    query,
+    setQuery,
+    items,
+    clearing,
+    processing,
+    isResultsLoaded,
+    activeMode,
+    resultsMode,
+    scrollDivRef,
+    onSubmit,
+    onQueryOpen,
+    onQueryDelete,
+    onAllQueriesDelete,
+    onQueryChangeMode,
+    onChangeGroupMode,
+    onQueryReRun,
+    onQueryProfile,
+  } = useQuery()
+
+  return (
+    <CreateIndexWrapper direction="column" justify="between">
+      <HeaderActions />
+
+      <StyledResizableContainer direction="vertical">
+        <ResizablePanel id="top-panel" minSize={20} defaultSize={30}>
+          <QueryWrapper
+            query={query}
+            activeMode={activeMode}
+            resultsMode={resultsMode}
+            setQuery={setQuery}
+            setQueryEl={() => {}}
+            onSubmit={onSubmit}
```

Comment:
There is something odd happening here, but submitting the query, it doesn't work for me. The query is not passed properly to the `handleSubmit` in the `useQuery` hook, and nothing happens. 

However, if I make this simple change, it works.

```
onSubmit={() => onSubmit()}
```